File: src/glider.cpp
#include "util.h"

#include "glider.h"

Glider::Glider() : Sprite("glider"), alive(false), origin{0, 0}, max(120.0) {}

void Glider::update(unsigned int ticks) {
	Sprite::update(ticks);

	if (Util::distance(origin, get_position()) > max)
		alive = false;
}

void Glider::create() {
	alive = true;

	origin = get_position();
}

void Glider::destroy() {
	alive = false;
}

File: behaviours/glider.lua
max = spec:get_float('glider/max')

function update (ticks)
	if math.sqrt((sprite.pos.x - projectile.origin.x)^2 + (sprite.pos.y - projectile.origin.y)^2) > max then
		projectile:kill()
	end

	if sprite.pos.x >= world.width - sprite.width or sprite.pos.x <= 0 or sprite.pos.y >= world.height - sprite.height or sprite.pos.y <= 0 then
		projectile:kill()
	end
end
